content/articles/web-automation-with-selenium/index.md

```diff
@@ -0,0 +1,107 @@
+```
+We also have to install a webdriver which is a tool that is needed for web automation. The web driver helps us interact with the browser. If you are using windows We will be using a windows package manager known as chocolatey to install the web driver, click [here]( https://chocolatey.org/install) to install chocolatey. To install we will use the command below
+```
+choco install chromdriver
+```
+If you are on a mac you the below command will do
+```
+brew cask install chromdriver
+```
+The chromedriver version should be compactable with your browser version but if in any way you encounter a compactable error, go to this [site](https://chromedriver.chromium.org/downloads) to download for your browser version and replace the one you have.
+
+Create a file `app.py` or whatever you will want to call your app and update it with the code below.
+
+```python
+from selenium import webdriver
+
+driver = webdriver.Chrome()
+
+driver.get('https://www.google.com/')
+```
+The above code snippet is used to open a browser and request a web `url`. The first line of code is importing webdriver from Selenium, in the second line we used the webdriver to open chrome which we stored in a variable `driver`.
+
+NOTE: There are different webdrivers for different browsers so if you prefer another browser just look for the webdriver for that browser on the internet, for instance, we would use `firefoxDriver` for the firefox browser. We are opening Chrome because we are using the `ChromeDriver`.
+
+On the third line, we used the driver variable to send a request to the browser requesting for the `url` in the parenthesis. You can run the code using the command below
+
+```
+python app.py
+```
+app is the name of our app file, if the name of your app is different yours will be different. If your code is correct the code will open your chrome browser as shown in the image below.
+
+![browser open image](/engineering-education/web-automation-with-selenium/image1.png)
+
+Next, we will be inputting a search keyword into the search field in the Google website. To do that we will have to get the search field element by inspecting the page. to inspect the page, right-click on the Google website page and click on inspect. The browser will open a window like shown in the image below
+
+![inspect browser](/engineering-education/web-automation-with-selenium/image2.png)
+
+There are different locator strategies to locate elements from the inspect element window this includes;
+- find_element_by_id 
+- find_element_by_name
+- find_element_by_class name
+- find_element_by_xpath
+
+We will be using xpath to get our elements in this guide but you can choose to use any of your choices. To get the element hover on the div tags and keep opening the one that highlights the search bar inclusive until you find the one that highlights only the search field then right-click on it, then click on copy and then click on copy xpath and paste it somewhere. Next, add the below lines to your code
```

Comment:
What's xpath? Why are we gonna use it?

Reply:
Always highlight keywords, variables, tags or codes within ` `` `.

For example, `div`, `xpath`